docs/reference/operators/kubeipresolver.md

```diff
@@ -0,0 +1,26 @@
+---
+title: KubeIPResolver
+---
+
+The KubeIPResolver operator enriches IP addresses with pod and service
+information. For instance, in this example the `10.96.0.10` IP is enriched with
+namespace, podname, kind, and pod labels.
+
+```json
+  "dst": {
+    "addr": "10.96.0.10",
+    "version": 4,
+    "namespace": "kube-system",
+    "podname": "kube-dns",
+    "kind": "svc",
+    "podlabels": {
+      "k8s-app": "kube-dns",
+      "kubernetes.io/cluster-service": "true",
+      "kubernetes.io/name": "CoreDNS"
+    }
+  }
+```
+
+## Parameters
+
+None
```

Comment:
> When is this operator enabled or disabled? I think each operator should document that.

I added a general comment indicating how operators are enabled or disabled. It's still missing to document that for each operator. I added a follow up on the PR description. 

> How does it relate to the flags --enrich-with-k8s-apiserver?

Good point. TBH I'm not sure, pinging @burak-ok since he has played with both things. 

> And when using the Golang API?

I suppose it'll work fine if there is a kubeconfig available, however I'm not sure how a user of the API could pass the clientset directly, I think users should be able to do that (later on).

Ref #3370

Reply:
The IP enricher currently is not working for image based gadgets. That seems to be an oversight. Issue https://github.com/inspektor-gadget/inspektor-gadget/issues/3483

The `enrich-with-k8s-apiserver` flag only adds `WithKubernetesEnrichment` and does nothing else. Enabling these operators (and passing the correct clientset) is still TODO. Issue https://github.com/inspektor-gadget/inspektor-gadget/issues/3484